sdk/storage/src/datalake/file_system_client.cpp

```diff
@@ -126,21 +138,35 @@ namespace Azure { namespace Storage { namespace DataLake {
         m_dfsUri.ToString(), *m_pipeline, options.Context, protocolLayerOptions);
   }
 
-  FileSystemGetMetadataResponse FileSystemClient::GetMetadata(
+  std::map<std::string, std::string> FileSystemClient::GetMetadata(
       const FileSystemGetMetadataOptions& options) const
   {
     DataLakeRestClient::FileSystem::GetPropertiesOptions protocolLayerOptions;
     // TODO: Add null check here when Nullable<T> is supported
     protocolLayerOptions.Timeout = options.Timeout;
     auto result = DataLakeRestClient::FileSystem::GetProperties(
         m_dfsUri.ToString(), *m_pipeline, options.Context, protocolLayerOptions);
-    return FileSystemGetMetadataResponse{
+    auto metadata = Details::DeserializeMetadata(result.Properties);
+    return metadata.HasValue() ? std::move(metadata.GetValue())
+                               : std::map<std::string, std::string>();
+  }
+
+  FileSystemProperties FileSystemClient::GetProperties(
+      const FileSystemPropertiesOptions& options) const
+  {
+    DataLakeRestClient::FileSystem::GetPropertiesOptions protocolLayerOptions;
+    // TODO: Add null check here when Nullable<T> is supported
+    protocolLayerOptions.Timeout = options.Timeout;
+    auto result = DataLakeRestClient::FileSystem::GetProperties(
+        m_dfsUri.ToString(), *m_pipeline, options.Context, protocolLayerOptions);
+    auto metadata = Details::DeserializeMetadata(result.Properties);
+    return FileSystemProperties{
         std::move(result.Date),
         std::move(result.ETag),
         std::move(result.LastModified),
         std::move(result.RequestId),
         std::move(result.Version),
-        Details::DeserializeMetadata(result.Properties),
+        metadata.HasValue() ? std::move(metadata.GetValue()) : std::map<std::string, std::string>(),
```

Comment:
Why not make metadata always non-nullable?

Reply:
Done.